packages/tables/src/Table/Concerns/HasQuery.php

```diff
@@ -144,9 +145,19 @@ public function getInverseRelationship(): ?string
             return null;
         }
 
-        return $this->evaluate($this->inverseRelationship) ?? (string) str(class_basename($relationship->getParent()::class))
-            ->plural()
-            ->camel();
+        $inverseRelationship = $this->evaluate($this->inverseRelationship);
+
+        if ($inverseRelationship) {
+            return $inverseRelationship;
+        }
+
+        $inverseRelationship = str(class_basename($relationship->getParent()::class));
+
+        if ($relationship instanceof HasMany) {
+            return (string) $inverseRelationship->singular()->camel();
```

Comment:
Can you give an example of how `singular()` on this line affects the output please? I assume the parent model class name to already be singular.

Also, I believe the check above should be for the `HasOneOrMany` class instead of just `HasMany`

Reply:
You're totally right about both!